packages/bot-web-ui/src/components/dashboard/dashboard.tsx

```diff
@@ -4,46 +4,76 @@ import { localize } from '@deriv/translations';
 import Chart from 'Components/chart';
 import DashboardComponent from './dashboard-component';
 import ReactJoyride from 'react-joyride';
-import { DBOT_ONBOARDING } from './joyride-config';
+import { DBOT_ONBOARDING, handleJoyrideCallback, getJoyrideToken } from './joyride-config';
+import classNames from 'classnames';
 import RunStrategy from './dashboard-component/run-strategy';
-import { connect } from 'Stores/connect';
-import RootStore from 'Stores/index';
 import RunPanel from '../run-panel';
 import QuickStrategy from './quick-strategy';
-import classNames from 'classnames';
+import { connect } from 'Stores/connect';
+import RootStore from 'Stores/index';
 import Tutorial from './tutorial-tab';
 
 type TDashboard = {
     active_tab: number;
     setActiveTab: (active_tab: number) => void;
+    onEntered: () => void;
+    has_file_loaded: boolean;
     toggleStrategyModal: () => void;
     is_drawer_open: boolean;
 };
 
-const Dashboard = ({ active_tab, setActiveTab, toggleStrategyModal, is_drawer_open }: TDashboard) => {
+const Dashboard = ({
+    active_tab,
+    setActiveTab,
+    toggleStrategyModal,
+    onEntered,
+    has_file_loaded,
+    is_drawer_open,
+}: TDashboard) => {
+    const [show_side_bar, setShowSideBar] = React.useState<boolean>(true);
     const [tour_run, setTourRun] = React.useState<boolean>(true);
+
     const handleClick = (e: React.MouseEvent) => {
         e.preventDefault();
         setTourRun(true);
         toggleStrategyModal();
     };
 
+    const [joyride_Token, setJoyrideToken] = React.useState<string>('');
+    React.useEffect(() => {
+        setJoyrideToken(getJoyrideToken());
+    }, [handleJoyrideCallback]);
```

Comment:
is `handleJoyrideCallback ` being used in this useEffect?

Reply:
actually i wanted to trigger the useeffect when the method handleJoyrideCallback triggers so i used it as a dependency